### PR TITLE
Remove plyr dependency (#116)

### LINKED ISSUE #116: Remove plyr dependency
It is mainly for `round_any()`, so could probably be removed, and would make scales a little more lightweight. Plus it removes a plyr dependency, which is probably a good thing in the long run.

### PR DESCRIPTION
**Pull request by dpseidel**
Removed plyr dependency to close #116. 

Each step is a separate commit. 
- removed `plyr` from DESCRIPTION 
- reimplemented `is.discrete` as it was only used in one location
- copied `round_any` method from `plyr`
- copied tests for `round_any` method
- rebuilt documentation

I chose not to export the `round_any` method, uncertain what best practice is when duplicating another package's function, so please look at that and let me know if you would prefer otherwise. 

And of course, if you think there is a wholly different or better way to reimplement `round_any`, let me know. 

A final note, when I rebuilt the documentation, there were a number of automatic changes to link formats in other .Rd files that I have not committed here to keep the PR clean but perhaps those will need to be included at some point.

### CODE REVIEW

File: R/scale-discrete.r [hunk 1]
```diff
@@ -22,7 +22,7 @@ dscale <- function(x, palette, na.value = NA) {
 train_discrete <- function(new, existing = NULL, drop = FALSE, na.rm = FALSE) {
   if (is.null(new)) return(existing)
 
-  if (!is.discrete(new)) {
```
Comment (hadley) [inline R/scale-discrete.r]: I think it's more readable to pull this out as a separate function.

File: R/round-any.r [hunk 2]
```diff
@@ -0,0 +1,21 @@
+#' Round to multiple of any number.
+#'
+#' @param x numeric or date-time (POSIXct) vector to round
+#' @param accuracy number to round to; for POSIXct objects, a number of seconds
+#' @param f rounding function: \code{\link{floor}}, \code{\link{ceiling}} or
+#'  \code{\link{round}}
+round_any <- function(x, accuracy, f = round) {
```
Comment (hadley) [inline R/round-any.r]: I'd say remove the documentation here.

File: R/round-any.r [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+#' @export
+round_any.numeric <- function(x, accuracy, f = round) {
+  f(x / accuracy) * accuracy
+}
+
+#' @export
+round_any.POSIXct <- function(x, accuracy, f = round) {
+  tz <- format(x[1], "%Z")
+  xr <- round_any(as.numeric(x), accuracy, f)
+  as.POSIXct(xr, origin="1970-01-01 00:00.00 UTC", tz=tz)
```
Comment (hadley) [inline R/round-any.r]: Can you please tidy style this?

File: tests/testthat/test-round-any.R [hunk 4]
```diff
@@ -0,0 +1,22 @@
+context("round_any")
+
+test_that("round_any() function", {
```
Comment (hadley) [inline tests/testthat/test-round-any.R]: Can you please split into two tests? One for numeric and one for POSIXct.

File: tests/testthat/test-round-any.R [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+  expect_equal(round_any(135, 100), 100)
+  expect_equal(round_any(135, 25), 125)
+  expect_equal(round_any(135, 10, floor), 130)
+  expect_equal(round_any(135, 100, floor), 100)
+  expect_equal(round_any(135, 25, floor), 125)
+  expect_equal(round_any(135, 10, ceiling), 140)
+  expect_equal(round_any(135, 100, ceiling), 200)
+  expect_equal(round_any(135, 25, ceiling), 150)
+
+  expect_equal(round_any(as.POSIXct("2000-01-01 11:00:00", tz="UTC"), 86400),
```
Comment (hadley) [inline tests/testthat/test-round-any.R]: Also needs to be tidy styled.

**Comment by dpseidel** [comment]
I made the suggested edits, as well as a few others to resolve warnings when building the documentation, and adopted your suggested commit style.

**Comment by hadley** [comment]
Can you move the documentation fixes into a separate branch so I can review them separately? You'll need to cherry-pick (or similar) the commits our of this branch and into a different one (@jimhester can help if you get stuck)

**Comment by dpseidel** [comment]
No problem. I'll figure it out.

**Comment by jimhester** [comment]
https://stackoverflow.com/a/28173964/2055486 gives a reasonably good strategy to do this

**Comment by dpseidel** [comment]
@jimhester Thanks so much! This was really helpful. Always a good day, when one learns some git tricks. 👍 

@hadley Things are clean now. I'll submit the documentation updates in a separate PR as requested.

**Pull request merged**